### PR TITLE
Clearing map when starting query

### PR DESCRIPTION
**Pull request by KakoozaJerry**

### CODE REVIEW

File: src/ldf-client-ui.js [hunk 1]
```diff
@@ -753,16 +753,19 @@ require('leaflet/dist/images/marker-shadow.png');
...
               }
             },
           }).addTo(self.map);
 
-          // Possibly rescale map view
-          self.map.fitBounds(self.mapLayer.getBounds());
+          self.trialLayerGroup.addLayer(laayer).addTo(self.map);
 
+          // Possibly rescale map view
+          // self.map.fitBounds(self.mapLayer.getBounds());
```
Comment (rubensworks) [inline src/ldf-client-ui.js:767]: Unneeded comments?

File: src/ldf-client-ui.js [hunk 2]
```diff
@@ -734,11 +735,10 @@ require('leaflet/dist/images/marker-shadow.png');
             geoFeature.properties.name = valueLabel.split('@', 1)[0].replace(/['"]+/g, '');
 
           // Add feature to map
-          L.geoJSON(geoFeature, {
+          let laayer = L.geoJSON(geoFeature, {
```
Comment (rubensworks) [inline src/ldf-client-ui.js:738]: Typo

File: src/ldf-client-ui.js [hunk 3]
```diff
@@ -734,11 +735,10 @@ require('leaflet/dist/images/marker-shadow.png');
...
             onEachFeature: function (feature) {
               // Add feature data to map
-              self.mapLayer.addData(feature);
-
+              // self.mapLayer.addData(feature);
```
Comment (rubensworks) [inline src/ldf-client-ui.js:741]: Unneeded comment

File: src/ldf-client-ui.js [hunk 4]
```diff
@@ -753,16 +753,19 @@ require('leaflet/dist/images/marker-shadow.png');
                 lat = feature.geometry.coordinates[1];
               }
               if (lon && lat) {
-                const marker = L.circleMarker([lat, lon]).addTo(self.map);
+                let marker = L.circleMarker([lat, lon]);
```
Comment (rubensworks) [inline src/ldf-client-ui.js:756]: Why is this change needed?

File: src/ldf-client-ui.js [hunk 5]
```diff
@@ -114,6 +114,7 @@ require('leaflet/dist/images/marker-shadow.png');
       // Initialize map
       this.map = L.map($map.get(0)).setView([52.517987721, 6.116665362], 8);
       this.mapLayer = L.geoJSON().addTo(this.map);
+      this.trialLayerGroup = L.layerGroup().addTo(this.map);
```
Comment (rubensworks) [inline src/ldf-client-ui.js:117]: What does `trialLayerGroup` mean? Can we have a more descriptive name?

File: src/ldf-client-ui.js [hunk 6]
```diff
@@ -114,6 +114,7 @@ require('leaflet/dist/images/marker-shadow.png');
       // Initialize map
       this.map = L.map($map.get(0)).setView([52.517987721, 6.116665362], 8);
       this.mapLayer = L.geoJSON().addTo(this.map);
```
Comment (rubensworks) [inline src/ldf-client-ui.js:116]: Will this one even be needed with these changed?

**Review by rubensworks — CHANGES_REQUESTED** [review]
Needs some cleanup

File: src/ldf-client-ui.js [hunk 7]
```diff
@@ -5,6 +5,7 @@
 window.jQuery = require('../deps/jquery-2.1.0.js');
 var N3 = require('n3');
 var resolve = require('relative-to-absolute-iri').resolve;
+var markerArray = [];
```
Comment (rubensworks) [inline src/ldf-client-ui.js:8]: Could we do this without global variables. This is considered bad programming practise.

Comment (KakoozaJerry) [reply src/ldf-client-ui.js:8]: I did make it such that it can collect all marker objects. I realised if I make it when I start executing the array it was not being recognized below in the part where I try to use it as I add markers immediately after adding them to the map

File: src/ldf-client-ui.js (hunk 6 above)
Comment (KakoozaJerry) [reply src/ldf-client-ui.js:117]: Yes, since I first add the features to a layer and then I add that layer to the LayerGroup. This helps me to add it to a layer on the map before its added to a layer group.

Comment (KakoozaJerry) [reply src/ldf-client-ui.js:117]: Yes, since I first add the features to a layer and then I add that layer to the LayerGroup. This helps me to add it to a layer on the map before its added to a layer group.

File: src/ldf-client-ui.js (hunk 5 above)
Comment (KakoozaJerry) [reply src/ldf-client-ui.js:118]: This is the LayerGroup variable am adding my layers to. sorry named it trial as I was testing it out

File: src/ldf-client-ui.js (hunk 4 above)
Comment (KakoozaJerry) [reply src/ldf-client-ui.js:758]: If you add the last bit of addTo(self.map) it will be added to the map outside a layer so ideally we need to add the marker to a layer.

Then the let instead of const, Thought const makes it fixed throughout yet the marker value keeps changing as you add various points.

File: src/ldf-client-ui.js (hunk 3 above)
Comment (KakoozaJerry) [reply src/ldf-client-ui.js:743]: Yes, sorry about that. we actually need to do away with that `self.mapLayer.addData(feature);` completely in the code. Shouldn't be part.

File: src/ldf-client-ui.js (hunk 2 above)
Comment (KakoozaJerry) [reply src/ldf-client-ui.js:740]: Sorry about this , needs to be mapLayer, should change it in my  next commit. Need a word that is not just layer such that it cannot conflict with the geoJSON features layers object

File: src/ldf-client-ui.js (hunk 1 above)
Comment (KakoozaJerry) [reply src/ldf-client-ui.js:772]: Yes, going to remove them in my next commit
